File: views/principaleView.py
import discord
from datetime import datetime
from utils.classGenerator import find_category_list
import random

class PrincipaleView(discord.ui.View):
    def __init__(self):
        super().__init__(timeout=300)

    @discord.ui.button(label="🔫 Fusils d'assaut", style=discord.ButtonStyle.primary, row=0)
    async def fusils_assaut(self, interaction: discord.Interaction, button: discord.ui.Button):
        # Réponse immédiate pour la réactivité
        await interaction.response.defer()
        
        armes = find_category_list("Fusils d'assaut")
        if not armes:
            return await interaction.followup.send("❌ Aucune arme dans cette catégorie", ephemeral=True)
        
        arme = random.choice(armes)
        embed = discord.Embed(title="🔫 Principale — Fusils d'assaut", description=f"```{arme}```", color=0x00ccff, timestamp=datetime.now())
        await interaction.edit_original_response(embed=embed, view=self)

    @discord.ui.button(label="🔫 Mitraillettes", style=discord.ButtonStyle.primary, row=0)
    async def mitraillettes(self, interaction: discord.Interaction, button: discord.ui.Button):
        await interaction.response.defer()
        
        armes = find_category_list("Mitraillettes")
        if not armes:
            return await interaction.followup.send("❌ Aucune arme dans cette catégorie", ephemeral=True)
        
        arme = random.choice(armes)
        embed = discord.Embed(title="🔫 Principale — Mitraillettes", description=f"```{arme}```", color=0x00ccff, timestamp=datetime.now())
        await interaction.edit_original_response(embed=embed, view=self)

    @discord.ui.button(label="🔫 Fusils à pompe", style=discord.ButtonStyle.primary, row=1)
    async def fusils_pompe(self, interaction: discord.Interaction, button: discord.ui.Button):
        await interaction.response.defer()
        
        armes = find_category_list("Fusils à pompe")
        if not armes:
            return await interaction.followup.send("❌ Aucune arme dans cette catégorie", ephemeral=True)
        
        arme = random.choice(armes)
        embed = discord.Embed(title="🔫 Principale — Fusils à pompe", description=f"```{arme}```", color=0x00ccff, timestamp=datetime.now())
        await interaction.edit_original_response(embed=embed, view=self)

    @discord.ui.button(label="🔫 Mitrailleuses", style=discord.ButtonStyle.primary, row=1)
    async def mitrailleuses(self, interaction: discord.Interaction, button: discord.ui.Button):
        await interaction.response.defer()
        
        armes = find_category_list("Mitrailleuses")
        if not armes:
            return await interaction.followup.send("❌ Aucune arme dans cette catégorie", ephemeral=True)
        
        arme = random.choice(armes)
        embed = discord.Embed(title="🔫 Principale — Mitrailleuses", description=f"```{arme}```", color=0x00ccff, timestamp=datetime.now())
        await interaction.edit_original_response(embed=embed, view=self)

    @discord.ui.button(label="🔫 Fusils tactiques", style=discord.ButtonStyle.primary, row=2)
    async def fusils_tactiques(self, interaction: discord.Interaction, button: discord.ui.Button):
        await interaction.response.defer()
        
        armes = find_category_list("Fusils tactiques")
        if not armes:
            return await interaction.followup.send("❌ Aucune arme dans cette catégorie", ephemeral=True)
        
        arme = random.choice(armes)
        embed = discord.Embed(title="🔫 Principale — Fusils tactiques", description=f"```{arme}```", color=0x00ccff, timestamp=datetime.now())
        await interaction.edit_original_response(embed=embed, view=self)

    @discord.ui.button(label="🔫 Fusils de précision", style=discord.ButtonStyle.primary, row=2)
    async def fusils_precision(self, interaction: discord.Interaction, button: discord.ui.Button):
        await interaction.response.defer()
        
        armes = find_category_list("Fusils de précision")
        if not armes:
            return await interaction.followup.send("❌ Aucune arme dans cette catégorie", ephemeral=True)
        
        arme = random.choice(armes)
        embed = discord.Embed(title="🔫 Principale — Fusils de précision", description=f"```{arme}```", color=0x00ccff, timestamp=datetime.now())
        await interaction.edit_original_response(embed=embed, view=self)

    @discord.ui.button(label="↩️ Retour", style=discord.ButtonStyle.secondary, row=3)
    async def retour(self, interaction: discord.Interaction, button: discord.ui.Button):
        # Réponse immédiate
        await interaction.response.defer()
        
        # Import local pour éviter les imports circulaires
        from views.rollView import RollView, create_class_embed
        from utils.classGenerator import generer_classe
        classe = generer_classe()
        embed = create_class_embed(classe)
        view = RollView(classe)
        await interaction.edit_original_response(embed=embed, view=view)

    @discord.ui.button(label='SAUVEGARDER', style=discord.ButtonStyle.success, emoji='💾', row=3)
    async def sauvegarder_arme_principale(self, interaction: discord.Interaction, button: discord.ui.Button):
        await interaction.response.defer(ephemeral=True)
        
        try:
            # Trouver le salon "classe" dans le serveur
            salon_classe = None
            for channel in interaction.guild.text_channels:
                if channel.name == "🔫・classe":
                    salon_classe = channel
                    break
            
            if not salon_classe:
                return await interaction.followup.send(
                    "❌ **Salon '🔫・classe' introuvable** - Assurez-vous qu'un salon textuel nommé '🔫・classe' existe sur ce serveur.",
                    ephemeral=True
                )
            
            # Récupérer l'arme depuis l'embed actuel
            embed_actuel = interaction.message.embeds[0]
            arme = embed_actuel.description.strip('```')
            titre_arme = embed_actuel.title
            
            # Créer l'embed pour l'arme sauvegardée
            embed = discord.Embed(
                title="💾 Arme Principale Sauvegardée",
                description=f"**Sauvegardée par {interaction.user.mention}**",
                color=0x00ff00,
                timestamp=datetime.now()
            )
            embed.add_field(name=titre_arme, value=f"```{arme}```", inline=False)
            embed.set_footer(text=f"Sauvegardée depuis #{interaction.channel.name}")
            
            # Envoyer l'arme dans le salon "classe"
            await salon_classe.send(embed=embed)
            
        except Exception as e:
            await interaction.followup.send(
                f"❌ **Erreur lors de la sauvegarde :** {str(e)}",
                ephemeral=True
            )

File: views/principaleView.js
import { ActionRowBuilder, StringSelectMenuBuilder, EmbedBuilder } from 'discord.js';
import { findCategoryList } from '../utils/classGenerator.js';
import { getRandomChoice } from '../utils/random.js';

/**
 * Create embed for primary weapons selection
 * @returns {EmbedBuilder} Discord embed
 */
export function createPrincipaleEmbed() {
    return new EmbedBuilder()
        .setTitle('🔫 Armes principales')
        .setDescription('```Choisissez une catégorie pour obtenir une arme aléatoire dedans.```')
        .setColor(0x00ccff)
        .setTimestamp();
}

/**
 * Create the primary weapons selection view
 * @returns {Object} Components for the message
 */
export function createPrincipaleView() {
    const selectMenu = new StringSelectMenuBuilder()
        .setCustomId('select_principale')
        .setPlaceholder('Choisissez une catégorie d\'arme principale')
        .addOptions([
            {
                label: 'Fusils d\'assaut',
                description: 'Armes polyvalentes pour tous types de combat',
                value: 'Fusils d\'assaut',
                emoji: '🔫'
            },
            {
                label: 'Mitraillettes',
                description: 'Armes rapides pour le combat rapproché',
                value: 'Mitraillettes',
                emoji: '🏃'
            },
            {
                label: 'Fusils à pompe',
                description: 'Armes dévastatrices à courte portée',
                value: 'Fusils à pompe',
                emoji: '💥'
            },
            {
                label: 'Mitrailleuses',
                description: 'Armes lourdes avec beaucoup de munitions',
                value: 'Mitrailleuses',
                emoji: '🔥'
            },
            {
                label: 'Fusils tactiques',
                description: 'Armes précises pour le combat à moyenne portée',
                value: 'Fusils tactiques',
                emoji: '🎯'
            },
            {
                label: 'Fusils de précision',
                description: 'Armes à longue portée pour les tirs précis',
                value: 'Fusils de précision',
                emoji: '🎭'
            }
        ]);

    const row = new ActionRowBuilder()
        .addComponents(selectMenu);

    return { components: [row] };
}

/**
 * Handle primary weapon selection interaction
 * @param {Interaction} interaction - Discord interaction
 */
export async function handlePrincipaleSelection(interaction) {
    const selectedCategory = interaction.values[0];
    const weapons = findCategoryList(selectedCategory);
    
    if (weapons.length === 0) {
        await interaction.reply({
            content: `❌ Aucune arme trouvée dans la catégorie "${selectedCategory}"`,
            ephemeral: true
        });
        return;
    }

    const selectedWeapon = getRandomChoice(weapons);
    
    const embed = new EmbedBuilder()
        .setTitle(`🔫 ${selectedCategory}`)
        .setDescription(`Votre arme principale aléatoire :`)
        .setColor(0x00ff00)
        .setTimestamp()
        .addFields({
            name: '🎯 Arme sélectionnée',
            value: `\`\`\`${selectedWeapon}\`\`\``,
            inline: false
        })
        .setFooter({ text: '🍀 Bonne chance !' });

    await interaction.update({ embeds: [embed], components: [] });
}